.github/workflows/deploy.yml

```diff
@@ -150,33 +219,123 @@ jobs:
-          docker push $REGISTRY/$REPOSITORY:$IMAGE_TAG
-          echo "IMAGE=$REGISTRY/$REPOSITORY:$IMAGE_TAG" >> $GITHUB_OUTPUT
+          cd apps/$SERVICE && pnpm run docker:build
+          docker tag novu-$SERVICE $REGISTRY/$REPOSITORY/$SERVICE:latest
+          docker tag novu-$SERVICE $REGISTRY/$REPOSITORY/$SERVICE:$IMAGE_TAG
+          docker push $REGISTRY/$REPOSITORY/$SERVICE:latest
+          docker push $REGISTRY/$REPOSITORY/$SERVICE:$IMAGE_TAG
 
   deploy:
     needs: [build, prepare-matrix]
     runs-on: ubuntu-latest
     strategy:
       matrix:
         env: ${{ fromJson(needs.prepare-matrix.outputs.env_matrix).environment }}
-        service: ${{ fromJson(needs.prepare-matrix.outputs.service_matrix).service }}
+        service: ${{ fromJson(needs.prepare-matrix.outputs.deploy_matrix) }}
+
+    environment: ${{ matrix.env }}
+    
     steps:
-      - name: Print Important Info
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: ${{ vars.AWS_REGION }}
+      
+      - name: Download task definition
+        env:
+          ECS_PREFIX: ${{ vars.ECS_PREFIX }}
+          TASK_NAME: ${{ matrix.service.task_name }}
         run: |
-          echo "Deploying ${{ matrix.service }} to ${{ matrix.env }}"
-          echo "Docker Image: ${{ needs.build.outputs.docker_image }}"
+          aws ecs describe-task-definition --task-definition ${ECS_PREFIX}-${TASK_NAME} \
+          --query taskDefinition > task-definition.json
+
+      - name: Render Amazon ECS task definition
+        id: render-web-container
+        uses: aws-actions/amazon-ecs-render-task-definition@39c13cf530718ffeb524ec8ee0c15882bcb13842
+        with:
+          task-definition: task-definition.json
+          container-name: ${{ vars.ECS_PREFIX }}-${{ matrix.service.container_name }}
+          image: ${{secrets.ECR_URI}}/${{ vars.ECR_PREFIX }}/${{ matrix.service.image }}:${{ github.sha }}
+      
+      - name: Deploy to Amazon ECS service
+        uses: aws-actions/amazon-ecs-deploy-task-definition@3e7310352de91b71a906e60c22af629577546002
+        with:
+          task-definition: ${{ steps.render-web-container.outputs.task-definition }}
+          service: ${{ vars.ECS_PREFIX }}-${{ matrix.service.service_name }}
+          cluster: ${{ vars.ECS_PREFIX }}-${{ matrix.service.cluster_name }}
+          wait-for-service-stability: true
+
+    
+  sentry_release:
+    needs: [deploy, prepare-matrix]
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        service: ${{ fromJson(needs.prepare-matrix.outputs.service_matrix).service }}
+    environment: ${{ fromJson(needs.prepare-matrix.outputs.env_matrix).environment[0] }}
+    
+    steps:  
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Get NPM Version
```

Comment:
We don't use NPM semver for the services. We use it only for NPM packages. That is, I suggest using the latest commit hash to pinpoint the latest deployed code.